lib/reek/configuration/app_configuration.rb

```diff
@@ -33,19 +33,24 @@ class AppConfiguration
       #                          { Reek::Smells::UtilityFunction => { "enabled" => false } } }
       #
       #
-      # @param options [OpenStruct]
-      #   e.g.: #<OpenStruct report_format =: text,
-      #                      location_format =: numbers,
-      #                      colored = true,
-      #                      smells_to_detect = [],
-      #                      config_file = #<Pathname:config/defaults.reek>,
-      #                      argv = [ "lib/reek/spec" ]>
-      def initialize(options = nil)
-        @directory_directives = {}
-        @default_directive    = {}
-        @exclude_paths        = []
-
-        load options
+      # @param  path [Pathname] the path to the config file
+      # @param  directory_directives [Hash] see above for an example
+      # @param  default_directive [Hash] see above for an example
+      # @param  exclude_paths [Array] see above for an example
+      def initialize(path: nil,
+                     directory_directives: nil,
+                     default_directive: nil,
+                     exclude_paths: nil)
+        if path && [directory_directives, default_directive, exclude_paths].any?
+          raise(ArgumentError,
+                'You can either pass a path or single configuration values but not both')
+        end
```

Comment:
I feel somewhat bad about this, as it border on control coupling: if it’s either-or, then maybe we should have two constructors, one that takes a path and one that takes the other three? (A theoretical use case for having both would be ‘parse this path but set `exclude_paths` to the provided override’, but I’m not sure whether it would be useful in practice.)

(Also: this does not seem to be tested in any way.) ;)